### DISCUSSION TITLE
Recommend system: Hệ thống gợi ý hay đề xuất?

### DISCUSSION THREAD

**Discussion by lkhphuc** [d2l-ai/d2l-vi#2885] [Terminology]
Như đề, hiện đang được dịch là `gợi ý`, giống với `suggest`.
Mình thấy `đề xuất` hợp hơn 1 tẹo, mọi người thấy sao?
cc: @davidnvq @cuongvng @minhduc0711 @rootonchair @quangnhat185 @LongNguyen1984

**Comment by quangvnai** [comment]
* Từ `đề xuất` thì mình chưa nghe bao giờ, vừa tìm hiểu thì có [Netflix](https://help.netflix.com/vi/node/100639) với [Amazon](https://aws.amazon.com/vi/getting-started/hands-on/semantic-content-recommendation-system-amazon-sagemaker/) dịch như vậy.
* Từ `gợi ý` thì thấy có một vài bài báo khoa học (bằng tiếng Việt) sử dụng, và đông đảo các tutorials cũng dùng theo.
Thôi làm cái vote đi :))

**Comment by lkhphuc** [comment]
Tại mình hiểu `gợi ý` có thể được dịch từ `suggest`, `hint`, tuỳ trường hợp còn là `remind`. Còn `đề xuất` thì hẹp hơn là `recommend`.
👍: đề xuất
❤️: gợi ý

**Comment by quangnhat185** [comment]
Mình thì thích từ `đề xuất` vì nó có vẻ formal hơn

**Comment by thanhcsf** [comment]
Chốt đề xuất của @lkhphuc nhé.

**Comment by faalkao** [comment]
Mình thấy "recommendation" dịch ra là "đề xuất " thì có vẻ không sát nghĩa tiếng Việt. Đề xuất là nêu ra ý kiến và hướng giải quyết cho 1 vấn đề, và chưa biết cái đề xuất đó có đúng hay sai, có tối ưu hay không hay được dùng hay không. Còn recommendation system ở đây ý là hệ thống đã được sử dụng và thấy bổ ích hợp lý cho nhiều trường hợp và mọi người khuyên dùng. Theo mình nên dịch là "hệ thống khuyến nghị". Nếu dân giã hơn thì dùng là "hệ thống gợi ý" nghe dân giã nhưng sát nghĩa. Tất nhiên có nhiều đáp án hơn, tuy nhiên để đưa ra 1 đáp án hoàn hảo thì không hề dễ. 
P/s: Khi dịch, có lẽ mình không nên quá xem trọng 1 từ đã được dịch như thế nào trước đó và lấy theo số đông, vì nhiều từ được dùng nhiều nhưng chưa chắc đã đúng. Mình cần bám sát "định nghĩa " của từng từ và hiểu rõ điều mình muốn truyền đạt tới người đọc. Vì hãy tưởng tượng, nếu người đọc không hiểu (người nước ngoài đi chẳng hạn, như mình đi học tiếng Anh đó) thì mình sẽ tìm đến định nghĩa của từ đó (tài liệu đáng tin cậy). Góp ý nhỏ hi vọng có ích cho các bạn.

**Reply by quangnhat185** [reply]
Chào bạn, trước hết mình cảm ơn những góp ý của bạn. Tuy nhiên mình lại thấy có vẻ bạn hơi cứng nhắc trong việc định nghĩa hai từ `đề xuất` và `gợi ý`. Theo [Wikitionary](https://vi.wiktionary.org/wiki/g%E1%BB%A3i_%C3%BD#Ti%E1%BA%BFng_Vi%E1%BB%87t), `gợi ý` được định nghĩa là `Gợi ra để tự suy nghĩ, tự có ý kiến hoặc tự quyết định làm một việc nào đó.` Ở đây không hề nhắc đến ý "**đã được sử dụng, có tối ưu hay không hay được dùng hay không?**". Theo mình nghĩ để lựa chọn sử dụng hai từ `gợi ý` và `đề xuất` thì liên quan đến ngữ cảnh nhiều hơn. Ví dụ:
- Hế thống này có khả năng `đề xuất` nhiều sản phẩm tiềm năng tới khách hàng hơn
- App này sẽ `gợi ý` những nhà hàng gần với yêu cầu của bạn nhất

Mình thấy hai từ này khi dùng đối lẫn nhau vẫn giữ được nghĩa gốc, cho nên khi dịch thì có thể flexible được

**Reply by faalkao** [reply]
Mình đồng ý với bạn từ "gợi ý" không sát nghĩa, và có 1 solution khác mình đưa ra là "khuyến nghị". Anyway, dịch tài liệu chuyên ngành không hề dễ, trước mình cũng dịch một số tài liệu và để đưa ra các từ đúng khá là đau đầu. Chúc các bạn dịch tốt.

**Reply by lkhphuc** [reply]
> Còn recommendation system ở đây ý là hệ thống đã được sử dụng và thấy bổ ích hợp lý cho nhiều trường hợp và mọi người khuyên dùng.

Từ `khuyến nghị` (giống kiểu `khuyên dùng`) đúng là hợp với định nghĩa mà bạn đưa ra ở trên. 
Nhưng định nghĩa này mình thấy không chính xác lắm với các mô hình ML hay được sử dụng. Định nghĩa `recommendâtion` ở trên giống kiểu `Kem đánh răng PS được các nha sĩ khuyên dùng`, còn `recommendation system` trong ML thì mình hay hiểu là hệ thống đưa ra các đề xuất về video, phim, nhạc cho người dùng lựa chọn dựa trên sở thích và ngữ cảnh. Vậy nên mình thấy dịch recommendation thành `khuyến nghị` trong ngữ cảnh này không hợp bằng `đề xuất` hay `gợi ý`.
